internal/backup_push_handler.go

```diff
@@ -57,17 +60,26 @@ func (err BackupFromOtherBD) Error() string {
 
 type BackupConfig struct {
 	uploader                  *WalUploader
-	archiveDirectory          string
+	pgVersion                 int
+	pgDataDirectory           string
 	backupsFolder             string
+	backupName                string
+	startLSN                  uint64
+	endLSN                    uint64
+	uncompressedSize          int64
+	compressedSize            int64
```

Comment:
BackupConfig is used to define prerequisites for starting the backup process. Regarding these new fields, we fetch them in the process of the backup, after the startBackup has been called. Maybe store them in another struct?

Reply:
I created a BackupCharacteristics